### ISSUE TITLE
R reticulate package to be used in Shiny application.

### ISSUE THREAD

**Issue by tanthiamhuat** [ranikay/shiny-reticulate-app#8]
I must thank the authors of R package "reticulate" for their wonderful work. Python code are easily run inside R/RStudio now.

However, my needs is to use R package "reticulate" inside Shiny application. I see in your documentation https://github.com/ranikay/shiny-reticulate-app that it is possible with no issue.

My problem is as follows:
I went through those steps, so that it is clear to explain the issues.
library(reticulate)
reticulate::use_virtualenv('/srv/shiny-server/MyApplication',required = TRUE)
reticulate::py_config()
python:         /usr/bin/python3
libpython:      /usr/lib/python3.6/config-3.6m-x86_64-linux-gnu/libpython3.6.so
pythonhome:     //usr://usr
version:        3.6.9 (default, Oct  8 2020, 12:12:24)  [GCC 8.4.0]
numpy:          /home/Celaya/.local/lib/python3.6/site-packages/numpy
numpy_version:  1.19.2
python versions found: 
 /usr/bin/python3
 /usr/bin/python
above does not show the virtual environment being created, although I saw some directories (bin,include,lib,lib64,share,pyvenv.cfg) being added to my directory '/srv/shiny-server/MyApplication'.

On the server side of the shiny application:
library(reticulate)
reticulate::use_virtualenv('/srv/shiny-server/MyApplication',required = TRUE)
source_python("/srv/shiny-server/MyApplication/Python/imports.py",envir = globalenv(),convert = TRUE)
Above will crash my shiny application with the below error:
Listening on http://127.0.0.1:43541
Warning: Error in py_run_file_impl: ModuleNotFoundError: No module named 'pandas'

Detailed traceback: 
  File "<string>", line 5, in <module>

It seesm not able to load in the python module 'pandas' properly.

However, if I run below a standalone R-script inside RStudio (not on the server side), it has no issue:
library(reticulate)
use_python("/usr/bin/python3")
source_python("/srv/shiny-server/MyApplication/Python/imports.py",envir = globalenv(),convert = TRUE)

See also https://community.rstudio.com/t/rmarkdown-with-python-script/84382/10 for the issue which I posted.
Thank You.

**Comment by ranikay** [comment]
Hi!

I recommend NOT running `library(reticulate)`, as this causes the `reticulate` package to try and initialize a version of Python to use automatically, which may not be the one you want. Just delete that line in your code. You can run `reticulate` functions like `reticulate::use_virtualenv()` _without_ running `library(reticulate)` first.

Did you install the `pandas` package inside the virtual environment? You need to do this with `reticulate::virtualenv_install('YOUR VIRTUAL ENV PATH', packages = c('pandas'), ignore_installed=TRUE)`

To summarize, it looks like the system Python is being initialized (and you have `pandas` installed on your machine which is why it doesn't error when you run it stand-alone) but what you need to do is 1) force Shiny to use your virtual environment by NOT running `library(reticulate)` and 2) make sure you install `pandas` successfully into the virtual environment you want the app to use.

**Comment by tanthiamhuat** [comment]
Perfect ANSWERS for the day. Thank you very much. It makes my day easier now. And after I virtualenv_install for the first time, I can actually comment it for the next session, else it will re-install again.
Cheers.
